pkg/cmd/workflow/view/view.go

```diff
@@ -0,0 +1,253 @@
+package view
+
+import (
+	"errors"
+	"fmt"
+	"net/http"
+	"path/filepath"
+	"regexp"
+	"strings"
+
+	"github.com/AlecAivazis/survey/v2"
+	"github.com/MakeNowJust/heredoc"
+	"github.com/cli/cli/api"
+	"github.com/cli/cli/internal/ghrepo"
+	"github.com/cli/cli/pkg/cmd/workflow/shared"
+	"github.com/cli/cli/pkg/cmdutil"
+	"github.com/cli/cli/pkg/iostreams"
+	"github.com/cli/cli/pkg/markdown"
+	"github.com/cli/cli/pkg/prompt"
+	"github.com/cli/cli/utils"
+	"github.com/spf13/cobra"
+)
+
+type ViewOptions struct {
+	HttpClient func() (*http.Client, error)
+	IO         *iostreams.IOStreams
+	BaseRepo   func() (ghrepo.Interface, error)
+
+	WorkflowID string
+	Web        bool
+
+	Prompt bool
+	Raw    bool
+}
+
+func NewCmdView(f *cmdutil.Factory, runF func(*ViewOptions) error) *cobra.Command {
+	opts := &ViewOptions{
+		IO:         f.IOStreams,
+		HttpClient: f.HttpClient,
+	}
+	cmd := &cobra.Command{
+		Use:    "view [<run-id>]",
+		Short:  "View a summary of a workflow run",
+		Args:   cobra.MaximumNArgs(1),
+		Hidden: true,
+		Example: heredoc.Doc(`
+		  # Interactively select a workflow to view
+		  $ gh workflow view
+
+		  # View a specific workflow
+		  $ gh workflow view 0451
+		`),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			// support `-R, --repo` override
+			opts.BaseRepo = f.BaseRepo
+
+			opts.Raw = !opts.IO.CanPrompt()
+
+			if len(args) > 0 {
+				opts.WorkflowID = args[0]
+			} else if !opts.IO.CanPrompt() {
+				return &cmdutil.FlagError{Err: errors.New("workflow ID required when not running interactively")}
+			} else {
+				opts.Prompt = true
+			}
+
+			if runF != nil {
+				return runF(opts)
+			}
+			return runView(opts)
+		},
+	}
+	cmd.Flags().BoolVarP(&opts.Web, "web", "w", false, "Open workflow in the browser")
+
+	return cmd
+}
+
+func runView(opts *ViewOptions) error {
+	c, err := opts.HttpClient()
+	if err != nil {
+		return fmt.Errorf("could not build http client: %w", err)
+	}
+	client := api.NewClientFromHTTP(c)
+
+	repo, err := opts.BaseRepo()
+	if err != nil {
+		return fmt.Errorf("could not determine base repo: %w", err)
+	}
+
+	var workflow *shared.Workflow
+
+	if opts.Prompt {
+		workflow, err = promptWorkflows(client, repo)
+		if err != nil {
+			return err
+		}
+	}
+
+	if workflow == nil {
```

Comment:
Would this be equivalent to `!opts.Prompt`? I think that stating the condition in that manner more clearly shows that this is the non-interactive path.